Application/EdFi.Ods.Common/Configuration/CacheSettings.cs

```diff
@@ -5,23 +5,50 @@
 
 using System;
 using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
 
 namespace EdFi.Ods.Common.Configuration
 {
     public class CacheSettings
     {
-        public string ExternalCacheProvider { get; set; }
+        public const string ProviderNameRedis = "Redis";
+        private readonly List<string> _availableExternalProviders = new() { ProviderNameRedis };
+
+        public string ExternalCacheProvider { get; set; } = string.Empty;
         public RedisCacheSettings Redis { get; set; }
-        public SqlServerCacheSettings SQLServer { get; set; }
         public DescriptorsCacheConfiguration Descriptors { get; set; } = new();
         public PersonUniqueIdToUsiCacheConfiguration PersonUniqueIdToUsi { get; set; } = new();
         public ApiClientDetailsConfiguration ApiClientDetails { get; set; } = new();
         public SecurityCacheConfiguration Security { get; set; } = new();
         public ProfilesCacheConfiguration Profiles { get; set; } = new();
         public OdsInstancesCacheConfiguration OdsInstances { get; set; } = new();
-        
+
         public TenantsCacheConfiguration Tenants { get; set; } = new();
 
+        /// <summary>
+        /// Validates the `ExternalCacheProvider` setting when any of the "use
+        /// cache" settings are true.
+        /// </summary>
+        /// <exception cref="InvalidConfigurationException" />
+        public void Validate()
+        {
+            var usingExternalCache = Descriptors.UseExternalCache || PersonUniqueIdToUsi.UseExternalCache || ApiClientDetails.UseExternalCache;
+            var externalProviderIsValid = _availableExternalProviders.Where(x => string.Equals(x, ExternalCacheProvider, StringComparison.OrdinalIgnoreCase)).Any();
+
+            if (usingExternalCache & !externalProviderIsValid)
+            {
+                throw new ConfigurationErrorsException($"External caching has been enabled, but the specified cache provider \"{ExternalCacheProvider}\" is not valid.");
+            }
+
+            var cacheProviderIsRedis = string.Equals(ProviderNameRedis, ExternalCacheProvider, StringComparison.OrdinalIgnoreCase);
+            var redisConfigNotProvider = string.IsNullOrWhiteSpace(Redis.Configuration);
```

Comment:
Accessing the `Redis.Configuration` property here causes a null reference exception when the cache settings section is not explicitly defined. Should a null conditional operator be added after `Redis`? A null reference exception is occurring here when running some tests because the appsettings.json files in EdFi.Ods.WebApi.IntegrationTests and EdFi.Ods.WebApi.CompositeSpecFlowTests do not have a cache configuration section explicitly defined.

Reply:
I added a default value for the `Redis` property. Locally that resolved the problem.